server/src/test/java/io/crate/integrationtests/PartitionedTableConcurrentIntegrationTest.java

```diff
@@ -441,4 +443,51 @@ public void testConcurrentPartitionCreationWithColumnCreationTypeMissMatch() thr
         // this query should never fail
         execute("select * from t1 order by x limit 50");
     }
+
+    @Test
+    public void test_stress_partition_deletion_and_creation() throws Exception {
+        execute(
+            "create table tbl (x int, p int) " +
+            "clustered into 2 shards partitioned by (p) with (number_of_replicas = 0)");
+        int numOpsPerThread = randomIntBetween(100, 300);
+        int numInsertThreads = randomIntBetween(5, 10);
+        int numDeleteThreads = randomIntBetween(5, 10);
+        Phaser phaser = new Phaser(numInsertThreads + numDeleteThreads);
```

Comment:
Why is this needed? especially for the inserts? Isn't `register()` required?

Reply:
The phaser is used to ensure all threads have started before they run their actual logic.

Not necessarily required, but helps increase the chance that requests execute concurrently

Reply:
thx! and you don't need to call `register()` as well? Just had a quick look in `Phaser`'s java doc.